### ISSUE TITLE
descriptor.proto imports

### ISSUE THREAD

**Issue by miedzinski** [stepancheg/rust-protobuf#224]
Is compiling google's descriptor.proto required? I see that rust-protobuf includes it (https://github.com/stepancheg/rust-protobuf/blob/940858bedc0d6de9bca9391057d306022c79ef3e/protobuf/src/descriptor.rs), but generated code doesn't reuse it - I have to compile it myself.

I am compiling https://github.com/SteamDatabase/GameTracking-CSGO/blob/70fc992132adc7be89ad5b909b6f513189fd3d09/Protobufs/steammessages.proto with

```bash
root=$(dirname $0)
protoc \
    -I=$root/game-tracking/Protobufs/ \
    --rust_out=$root/src/proto \
    $root/game-tracking/Protobufs/steammessages.proto
```

and the generated code contains

```rust
pub mod exts {
    use protobuf::Message as Message_imported_for_functions;

    pub const key_field: ::protobuf::ext::ExtFieldOptional<super::super::descriptor::FieldOptions, ::protobuf::types::ProtobufTypeBool> = ::protobuf::ext::ExtFieldOptional { field_number: 60000, phantom: ::std::marker::PhantomData };

    pub const msgpool_soft_limit: ::protobuf::ext::ExtFieldOptional<super::super::descriptor::MessageOptions, ::protobuf::types::ProtobufTypeInt32> = ::protobuf::ext::ExtFieldOptional { field_number: 60000, phantom: ::std::marker::PhantomData };

    pub const msgpool_hard_limit: ::protobuf::ext::ExtFieldOptional<super::super::descriptor::MessageOptions, ::protobuf::types::ProtobufTypeInt32> = ::protobuf::ext::ExtFieldOptional { field_number: 60001, phantom: ::std::marker::PhantomData };
}
```

which obviously fails when passed to rustc, because I didn't provide descriptor.rs. I can easily add it, but is it intended?

**Comment by stepancheg** [comment]
I not 100% sure, but I think when rust-protoc compiles imports to `descriptor.proto`, it should use `descriptor.rs` from protobuf library (i. e. import it as `::protobuf::descriptor`), same way it does for [well known types](https://github.com/stepancheg/rust-protobuf/blob/940858bedc0d6de9bca9391057d306022c79ef3e/protobuf/src/codegen/rust_types_values.rs#L416).

**Comment by miedzinski** [comment]
Clearly it doesn't, at least with extensions. I've found offending function call - https://github.com/stepancheg/rust-protobuf/blob/940858bedc0d6de9bca9391057d306022c79ef3e/protobuf/src/codegen/extensions.rs#L60, but I'm not familiar with code base enough to fix it.

Replacing `super::super::descriptor` with `::protobuf::descriptor` by hand produces valid (and working) Rust code, so I guess it should be supported.

**Comment by miedzinski** [comment]
Thank you for such a quick fix, you're wonderful! :) Can I expect 1.3.2 release soon or should I run git version?

**Comment by stepancheg** [comment]
I think it should 1.4.0, because change is somewhat backward incompatible. I think I'll publish it later today.

**Comment by stepancheg** [comment]
Published 1.4.0.
